lib/msf/core/payload/windows/meterpreter_loader.rb

```diff
@@ -81,9 +81,9 @@ def generate_config(opts={})
       uuid:              opts[:uuid],
       transports:        opts[:transport_config] || [transport_config(opts)],
       extensions:        [],
-      stageless:         opts[:stageless] == true
+      stageless:         opts[:stageless] == true,
+      log_path:          Msf::OptMeterpreterDebugLogging.parse_logging_options(ds['MeterpreterDebugLogging'])[:rpath]
```

Comment:
This seems a bit clunky, should we consider having `Msf::OptMeterpreterDebugLogging` set object accessors when parsed during validation so the payload writers are not thinking about parsing every time?

This may require adding a `to_s` to the Opt class to support displaying current datastore setting.

I envision the `datastore` value could then be accessed as:
```suggestion
      log_path:          ds['MeterpreterDebugLogging'].rpath
```

Reply:
This could also simplify the existing usage of that parser method from the previous PR.
https://github.com/rapid7/metasploit-framework/blob/cb9e4121381430a08fd4b913a14347cd62bc6bec/lib/msf/core/payload/python/meterpreter_loader.rb#L77-L78
could become:
```
met.sub!(%q|DEBUGGING_LOG_FILE_PATH = None|, %Q|DEBUGGING_LOG_FILE_PATH = "#{ds['MeterpreterDebugLogging'].rpath}"|) if ds['MeterpreterDebugLogging'].rpath
```

Reply:
These comments still apply, unrelated changes marked then as outdated.

Reply:
@jmartin-r7 would this be a blocker for you? or would you be cool with coming back to this later just so we can get this shipped
I just think placing a more complex object into the datastore could lead to some...unexpected interactions that could delay this further

Reply:
>defines ds passed as opts[:datastore] as an override to the global datastore.
That's certainly true in this case but not for all of the payloads looking here for example https://github.com/rapid7/metasploit-framework/blob/7bfb1fbc14bbdbbe6cb41a2391185d23932ea926/modules/payloads/singles/windows/meterpreter_reverse_https.rb#L50-L60
it's not an option and likely where the pattern was picked up from for `datastore['MeterpreterDebugBuild']` initially

I agree though makes sense to parse these values out earlier in the function and pass them in, happy to do that